app/visualisation_methods/sensitivity/plugin.rb

```diff
@@ -0,0 +1,212 @@
+require 'json'
+
+class Sensitivity
+  attr_accessor :experiment
+  attr_accessor :parameters
+
+  attr_reader :series_to_plot
+  attr_reader :parameters_names
+  attr_reader :sorted_parameters_names
+  attr_reader :name_of_output
+  attr_reader :series_names
+  attr_reader :list_of_methods
+  attr_reader :method_name
+
+  def prepare_sensitivity_chart_content
+    output = "<script>(function() { \nvar i=" + parameters["chart_id"] + ";"
+    output += "\nvar sorted_parameters_names = " + @sorted_parameters_names.to_json + ";"
+    output += "\nvar series_to_plot = " +  @series_to_plot.to_json + ";"
+    output += "\nvar method_name = " +  @method_name.to_json + ";"
+    output += "\nsensitivity_main(i, series_to_plot, sorted_parameters_names, method_name);"
+    output += "\n})();</script>"
+    output
+  end
+
+  ##
+  # We assume that the result json have specific format:
+  #   morris method:
+  #   {
+  #     "sensitivity_analysis_method":"morris",
+  #     "moes":{
+  #       "moe1":{
+  #         "parameter1":{
+  #           "mean":3,
+  #           "st.deviation ":2
+  #         },
+  #         "parameter2":{
+  #           "mean":-1,
+  #           "st.deviation ":4
+  #         }
+  #       },
+  #       "moe2":{
+  #         "parameter1":{
+  #           "mean":5,
+  #           "st.deviation ":1
+  #          },
+  #          "parameter2":{
+  #           "mean":0,
+  #           "st.deviation ":2
+  #          }
+  #       }
+  #     }
+  #    }
+  #
+  #   fast method:
+  #   {
+  #     "sensitivity_analysis_method":"fast",
+  #     "moes":{
+  #       "moe2":{
+  #         "parameter1":{
+  #           "first_order":0.21,
+  #           "total_order":0.45
+  #         },
+  #         "parameter2":{
+  #           "first_order":0.32,
+  #           "total_order":0.23
+  #         }
+  #       },
+  #       "moe1":{
+  #         "parameter1":{
+  #           "first_order":0.23,
+  #           "total_order":0.34
+  #         },
+  #         "parameter2":{
+  #           "first_order":0.31,
+  #           "total_order":0.64
+  #         }
+  #       }
+  #     }
+  #   }
+
+  def handler()
+    if parameters["id"] && parameters["chart_id"] && parameters["output"]
+      @name_of_output = parameters["output"]
+      @list_of_methods = ["morris", "fast"]
+      output_hash = @experiment.results
+
+      if !@list_of_methods.include? output_hash["sensitivity_analysis_method"]
+        raise("No visualization for #{output_hash["sensitivity_analysis_method"]} method")
+      end
+
+      if @experiment.error_reason != nil
+        raise ("Error appeared - #{@experiment.error_reason}")
```

Comment:
Wydaje mi się, że w tym przypadku można napisać coś bardziej konkretnego. Np. "The experiment ended with error: ..."